insights/combiners/netstat.py

```diff
@@ -0,0 +1,100 @@
+"""
+Link Layer stats
+================
+
+Combiner for link layer stats. It uses the results of
+the ``netstat -i`` parser and the ``ip -s link`` parser to determine the
+network stats of link layer. ``ip -s link`` is the preferred source
+of data and return object which can be used to access ``group_by_iface``.
+
+Examples:
+    >>> nstat = shared[NetworkStats]
+    >>> stats = nstat.group_by_iface
+    >>> stats["lo"]
+    {'RX-OK': 98, 'TX-OK': 100, 'MTU': 65536, 'RX-ERR': 0, 'TX-DRP': 0, 'TX-ERR': 0, 'RX-DRP': 0, 'RX-OVR': 0, 'Flg': 'LRU'}
+    >>> stats["enp0s8"]
+    {'RX-OK': 6, 'TX-DRP': 0, 'TX-OK': 4, 'MTU': 1500, 'RX-ERR': 0, 'TX-ERR': 0, 'RX-DRP': 0, 'RX-OVR': 0, 'Flg': 'BMRU'}
+    >>> print nstat.data[0]['Iface']
+    enp0s8
+    >>> for dev_item in nstat.data:
+    >>>     print dev_item
+    {'RX-OK': '842447', 'TX-OVR': '0', 'Iface': 'bond1', 'TX-OK': '4233', 'MTU': '1500', 'Met': '0', 'RX-ERR': '0', 'TX-DRP': '0', 'TX-ERR': '0', 'RX-DRP': '0', 'RX-OVR': '0', 'Flg': 'BMmRU'}
+    {'RX-OK': '422518', 'TX-OVR': '0', 'Iface': 'eth0', 'TX-OK': '1703', 'MTU': '1500', 'Met': '0', 'RX-ERR': '0', 'TX-DRP': '0', 'TX-ERR': '0', 'RX-DRP': '0', 'RX-OVR': '0', 'Flg': 'BMsRU'}
+    ...
+    ...
+"""
+
+from insights.core.plugins import combiner
+from insights.parsers.netstat import Netstat_I
+from insights.parsers.ip import IpLinkInfo
+from copy import deepcopy
+HANDLD_FLAGS = ["LOWER_UP", "NOARP", "NO-CARRIER"]
+
+
+@combiner(requires=[IpLinkInfo, Netstat_I])
+class NetworkStats(object):
+    """
+    A combiner for working with both `netstat -i` and `ip -s link` That is if
+    ``netstat -i`` gets deprecated then we can continue with ``ip -s link``.
+
+    This interface closely follow interface Netstat_I it has following property:
+
+    *   ``group_by_iface`` this property returns network stats organised per interface
+        in dict format.
+
+    *   ``self.stats`` (dict) dictionary of network stats as per devices.
+
+    *   ``self.data`` (list) list of network stats as per devices.
+    """
+
+    def __init__(self, local, shared):
+        if Netstat_I in shared:
+            netstat = shared[Netstat_I]
+            self.data = netstat.data
+            ldata = deepcopy(netstat.data)
+            self.group_by_iface = self._group_by_iface(ldata)
+        elif IpLinkInfo in shared:
+            nstat = shared.get(IpLinkInfo)
+            self.group_by_iface, self.data = self._organize_for_netstat(nstat.data)
+
+    def _group_by_iface(self, ldata):
+        periface = {}
+        for ifdata in ldata:
+            iface_name = ifdata["Iface"]
+            del ifdata["Iface"]
+            periface[iface_name] = ifdata
+        return periface
+
+    def _get_flags(self, v):
```

Comment:
1. Since this `_get_flags` are only called by "_organize_for_netstat", better move it inside the "_organize_for_netstat" as a inner private method. 
2. For the 2nd parameter, better pass the v['flags'] directly instead of the whole `dict` v"